core/src/main/scala/kafka/server/KafkaRequestHandler.scala

```diff
@@ -227,7 +227,7 @@ class KafkaRequestHandlerPool(val brokerId: Int,
   }
 }
 
-class BrokerTopicMetrics(name: Option[String]) {
+class BrokerTopicMetrics(name: Option[String], systemRemoteStorageEnabled: Boolean) {
```

Comment:
In it's current form, this is not very extensible for future cases because if we want to optionally add metrics for a new feature, we will have to add new booleans as arguments to determine the usage of that feature. 

Perhaps we can either pass the entire config object and let this `BrokerTopicMetrics` class make a decision based on config on what metrics it want to enable.

Reply:
Passing `config` now. Thanks.